datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1161,6 +1161,12 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
                 lit(!negated)
             }
 
+            // a IS NOT NULL --> true, if a is not nullable
+            Expr::IsNotNull(expr) if !info.nullable(&expr)? => lit(true),
```

Comment:
Thanks @byteink. would you mind adding more description? Why the simplification needed and what exactly problem it solves? is it a bug or something?

Reply:
```sql
create table test(a int not null);

select a is null
```

we can convert it into 

```sql
select false
```

Reply:
@comphead It is not a bug. I've added some descriptions.